src/server/scripts/Outland/zone_hellfire_peninsula.cpp

```diff
@@ -420,10 +420,480 @@ class npc_fel_guard_hound : public CreatureScript
+                            case 15:
+                                Talk(SAY_BARADA_6);
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 16:
+                                if (jules)
+                                    jules->AI()->Talk(SAY_JULES_5);
+
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 17:
+                                Talk(SAY_BARADA_7);
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 18:
+                                if (jules)
+                                    jules->AI()->Talk(SAY_JULES_3);
+
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 19:
+                                Talk(SAY_BARADA_7);
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 20:
+                                if (jules)
+                                {
+                                    jules->GetAI()->DoAction(ACTION_JULES_MOVE_HOME);
+                                    jules->RemoveAura(SPELL_JULES_VOMITS_AURA);
+                                }
+
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 21:
+                                //End
+                                if (playerGUID)
+                                {
+                                    if (Player* player = ObjectAccessor::FindPlayer(playerGUID))
+                                        player->KilledMonsterCredit(NPC_COLONEL_JULES, ObjectGuid::Empty);
+                                }
+
+                                if (jules)
+                                    jules->RemoveAllAuras();
+
+                                me->RemoveAura(SPELL_BARADAS_COMMAND);
+                                me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PACIFIED);
+
+                                Talk(SAY_BARADA_8);
+                                me->GetMotionMaster()->MoveTargetedHome();
+                                EnterEvadeMode();
+                                break;
+                        }
+                        break;
+                }
+            }
+        }
+
+        private:
+        EventMap events;
+        uint8 step;
+        Creature* jules;
```

Comment:
You mean like  ccrs said in https://github.com/TrinityCore/TrinityCore/pull/14412#discussion_r27441065 "never store a pointer, get its objectguid" ? Could you show me an example of how to get that OGUID? :mag: :memo: